### PR TITLE
Add resources to deploy to RHOAI workspaces

### PR DESCRIPTION
**Pull request by johnmcollier**
Fixes https://issues.redhat.com/browse/DEVHAS-691

- Adds in the set of Kubernetes resources needed to deploy the sample applications to OpenShift AI
- Includes an OpenShift Project resource to deploy the notebook under

### CODE REVIEW

File: templates/http/base/kustomization.yaml [hunk 1]
```diff
@@ -16,3 +16,6 @@ resources:
 - deployment.yaml
 - route.yaml
 - service.yaml
+- {%- if values.rhoaiSelected %}
```
Comment (yangcao77) [inline templates/http/base/kustomization.yaml]:
```suggestion
{%- if values.rhoaiSelected %}
```

File: templates/http/base/rhoai/dsp.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+kind: Project
+apiVersion: project.openshift.io/v1
+metadata:
+  name: {{values.dsp}}
+  labels:
+    kubernetes.io/metadata.name: {{values.dsp}}
+    opendatahub.io/dashboard: 'true'
+  annotations:
+    openshift.io/description: ''
```
Comment (yangcao77) [inline templates/http/base/rhoai/dsp.yaml]: do we need this if it's empty?

File: templates/http/base/rhoai/notebook.yaml [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+kind: Notebook
+metadata:
+  annotations:
+    notebooks.opendatahub.io/inject-oauth: 'true'
+    opendatahub.io/image-display-name: Minimal Python
+    openshift.io/display-name: {{values.name}}
+  name: {{values.name}}
+  namespace: {{values.dsp}}
+  labels:
+    app: {{values.name}}
```
Comment (yangcao77) [inline templates/http/base/rhoai/notebook.yaml]: do we need this?  or the parent app is set already by `base/kustomization.commonLabels`

File: templates/http/base/rhoai/pvc.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  annotations:
+    openshift.io/description: ''
```
Comment (yangcao77) [inline templates/http/base/rhoai/pvc.yaml]: same question on the empty description

**Review by yangcao77 — COMMENTED** [review]
looks good to me, just some minor questions

File: templates/http/base/rhoai/notebook.yaml (hunk 3 above)
Comment (johnmcollier) [reply templates/http/base/rhoai/notebook.yaml]: Huh, TIL, if it's set by the common labels, then no, it's not needed.

File: templates/http/base/rhoai/dsp.yaml (hunk 2 above)
Comment (johnmcollier) [reply templates/http/base/rhoai/dsp.yaml]: Nope, I'll remove it

File: templates/http/base/rhoai/notebook.yaml [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+            initialDelaySeconds: 10
+            periodSeconds: 5
+            successThreshold: 1
+            timeoutSeconds: 1
+          env:
+            - name: NOTEBOOK_ARGS
+            - name: JUPYTER_IMAGE
+              value: 'image-registry.openshift-image-registry.svc:5000/redhat-ods-applications/s2i-minimal-notebook:2024.1'
+            - name: MODEL_ENDPOINT
+              value: http://{{values.name}}-model-server:{{values.modelServicePort}}
```
Comment (yangcao77) [inline templates/http/base/rhoai/notebook.yaml]: QQ: the model pod deployed via `http/base/deployment-model-server.yaml` is in a different namespace. is the endpoint still valid and able to connect?

Comment (johnmcollier) [reply templates/http/base/rhoai/notebook.yaml]: Hmm, that was the one thing I still needed to test today. Currently, the URL present wouldn't work. However, we can access the service endpoint from any namespace with the following:
```
<service-name>.<namespace>.svc.cluster.local
```

We already have the service name via `{{values.name}}-model-server`, but I still need to figure out how to programmatically get the namespace from the template. Would `{{values.namespace}}` work?

File: templates/http/base/rhoai/pvc.yaml (hunk 4 above)
Comment (johnmcollier) [reply templates/http/base/rhoai/pvc.yaml]: Updated

File: templates/http/base/rhoai/notebook.yaml (hunk 3 above)
Comment (johnmcollier) [reply templates/http/base/rhoai/notebook.yaml]: Updated to remove the label

File: templates/http/base/rhoai/dsp.yaml (hunk 2 above)
Comment (johnmcollier) [reply templates/http/base/rhoai/dsp.yaml]: Updated

File: templates/http/base/rhoai/notebook.yaml (hunk 5 above)
Comment (yangcao77) [reply templates/http/base/rhoai/notebook.yaml]: yes, `{{values.namespace}}` is the namespace deploys the model server.

Comment (yangcao77) [reply templates/http/base/rhoai/notebook.yaml]: so it should be  `{{values.name}}-model-server.{{values.namespace}}.svc.cluster.local:{{values.modelServicePort}}`?

Comment (johnmcollier) [reply templates/http/base/rhoai/notebook.yaml]: Nice, I’ll go update the template then

Comment (johnmcollier) [reply templates/http/base/rhoai/notebook.yaml]: Updated

**Review by yangcao77 — APPROVED** [review]
looks good to me

**Pull request merged**